# HG changeset patch
# User Torsten <ttl@justmail.de>
# Date 1380820140 -7200
#      Thu Oct 03 19:09:00 2013 +0200
# Node ID dbda6edfe01d877939459b97655b7dbbafa04bb0
# Parent  93362ad454b9c176dc84106b96c1c684de7f028e
* octave-gui.cc: install translators before showing the wlecome wizard

diff --git a/libgui/src/octave-gui.cc b/libgui/src/octave-gui.cc
--- a/libgui/src/octave-gui.cc
+++ b/libgui/src/octave-gui.cc
@@ -89,33 +89,33 @@ dissociate_terminal (void)
 
 int
 octave_start_gui (int argc, char *argv[])
 {
   dissociate_terminal ();
 
   QApplication application (argc, argv);
 
+  // install translators for the gui and qt text
+  QTranslator gui_tr, qt_tr, qsci_tr;
+  resource_manager::config_translators (&qt_tr,&qsci_tr,&gui_tr);
+  application.installTranslator (&qt_tr);
+  application.installTranslator (&qsci_tr);
+  application.installTranslator (&gui_tr);
+
   while (true)
     {
       if (resource_manager::is_first_run ())
         {
           welcome_wizard welcomeWizard;
           welcomeWizard.exec ();
           resource_manager::reload_settings ();
         }
       else
         {
-          // install translators for the gui and qt text
-          QTranslator gui_tr, qt_tr, qsci_tr;
-          resource_manager::config_translators (&qt_tr,&qsci_tr,&gui_tr);
-          application.installTranslator (&qt_tr);
-          application.installTranslator (&qsci_tr);
-          application.installTranslator (&gui_tr);
-
           // update network-settings
           resource_manager::update_network_settings ();
 
 #if ! defined (__WIN32__) || defined (__CYGWIN__)
           // If we were started from a launcher, TERM might not be
           // defined, but we provide a terminal with xterm
           // capabilities.
 
